src/app/internal/guiapp.cpp

```diff
@@ -149,8 +164,21 @@ void GuiApp::perform()
     const QUrl url(QStringLiteral("qrc:/qml") + mainQmlFile);
 #endif
 
-    QObject::connect(engine, &QQmlApplicationEngine::objectCreated,
-                     qApp, [this, url, splashScreen](QObject* obj, const QUrl& objUrl) {
+    QObject::connect(engine, &QQmlApplicationEngine::objectCreated, qApp,
+                     [this, url, splashScreen, glProblemsDetector](QObject* obj, const QUrl& objUrl) {
+        if (glProblemsDetector->isProblemWithGl()) {
```

Comment:
It seems that if this condition is false, the `glProblemsDetector` will not be deleted

Reply:
fixed